### ISSUE TITLE
Build for a single target fails with "go_first_class" in targets

### ISSUE THREAD

**Issue by hansmi** [goreleaser/goreleaser#4426]
### What happened?

Commit e33d0536129a added an error when a `--single-target` build finds no target:

```
no builds matching --single-target linux/amd64
```

This works when `builds[].targets` is left at its implicit default value or includes the selected target. The error above is from a build with `go_first_class` in the targets.

```yaml
builds:
  - […]
    targets:
      - go_first_class
```

From the documentation:

https://github.com/goreleaser/goreleaser/blob/52d976beaea9ba8a5b0f09f90f22135b468c33c2/www/docs/customization/builds.md?plain=1#L152-L155

### How can we reproduce this?

Preparation:

```shell
$ git clone https://github.com/hansmi/prombackup.git
$ cd prombackup
$ git reset --hard 8985b66fd90a2a1f2764
```

Reproduction:

```shell
$ docker run --rm -v $PWD:/go/src/github.com/user/repo -w /go/src/github.com/user/repo goreleaser/goreleaser build --snapshot --clean --single-target
  • starting build...
  • loading                                          path=.goreleaser.yml
  • building only for linux/amd64                    reason=single target is enabled
  ⨯ build failed after 0s                            error=no builds matching --single-target linux/amd64
```

Demonstration that `go_first_class` is to blame:

```shell
$ sed -i -e '/go_first_class/ d' .goreleaser.yml
$ docker run --rm -v $PWD:/go/src/github.com/user/repo -w /go/src/github.com/user/repo goreleaser/goreleaser build --snapshot --clean --single-target
  • starting build...
  • loading                                          path=.goreleaser.yml
  • building only for linux/amd64                    reason=single target is enabled
  • skipping validate...
  • loading environment variables
  • getting and validating git state
    • git state                                      commit=8985b66fd90a2a1f276463571b332bf7f8522527 branch=main current_tag=v0.0.6 previous_tag=v0.0.5 dirty=true
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
  • snapshotting
    • building snapshot...                           version=0.0.7-snapshot20231115T221630Z+g8985b66
  • checking distribution directory
    • cleaning dist
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/prombackup-server_linux_amd64_v1/prombackup-server
    • building                                       binary=dist/prombackup_linux_amd64_v1/prombackup
    • took: 9s
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • build succeeded after 9s
  • thanks for using goreleaser!
```

### goreleaser version

```bash
$ docker run --rm -v $PWD:/go/src/github.com/user/repo -w /go/src/github.com/user/repo goreleaser/goreleaser --version
[…]
GitVersion:    1.22.1
GitCommit:     52d976beaea9ba8a5b0f09f90f22135b468c33c2
GitTreeState:  false
BuildDate:     2023-11-08T23:40:43Z
BuiltBy:       goreleaser
GoVersion:     go1.21.4
Compiler:      gc
ModuleSum:     h1:7+P9DHpKPmg160mni77AC6x43tch0DKNqULIfrYRyOU=
Platform:      linux/amd64
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
this should also be fixed by #4442 

can you test that PR?

**Comment by hansmi** [comment]
@caarlos0, I can confirm that  #4442 resolves the error reported in this issue. Thank you!
